src/main/java/com/mixpanel/android/mpmetrics/ResourceReader.java

```diff
@@ -16,10 +19,9 @@
  */
 public abstract class ResourceReader implements ResourceIds {
 
-    public static class Ids extends ResourceReader {
-        public Ids(String resourcePackageName, Context context) {
-            super(context);
-            mResourcePackageName = resourcePackageName;
+    public static class Id extends ResourceReader {
+        public Id(String resourcePackageName, Context context) {
+            super(resourcePackageName, context);
```

Comment:
Since you are doing this refactoring, it might make sense to do something like:

```
public interface ResourceIds {
    public void initialize();
    ...
}
```

and have the main class call that method that `Id` and `Drawable` need to override. So:

```
protected ResourceReader(String resourcePackageName, Context context) {
    mResourcePackageName = resourcePackageName;
    mContext = context;	        mContext = context;
    mIdNameToId = new HashMap<String, Integer>();
    mIdToIdName = new SparseArray<String>()
    initialize();
}
```

so we can remove these constructors

Reply:
If we remove these constructors, won't these subclasses look for the default constructor in ResourceReader which is undefined since we wrote an explicit constructor so the implicit default constructor isn't generated? If we explicitly define the default constructor, how will we capture the resourcePackageName and context args? I don't see a way to remove these constructors with the desired effect

Reply:
you're right, there is no default constructor. a simple thing is to just call super
```
public Ids(String name, Context ctx) {
    super(name,ctx);
}
```

Reply:
Do we still want to make initialize part of the interface and have sub classes of ResourceReader implement them given that the functionality is the same and we can't remove these constructors anyways since we need to call super?